### ISSUE TITLE
[BUG]  :w write paste wl-clipboard contents  

### ISSUE THREAD

**Issue by nonumeros** [mawww/kakoune#5160]
### Version of Kakoune

2024.04.25

### Reproducer

```
copy
foo-'bar-baz'-foo
```

on the ~scratch~ minibuffer  `:w` and paste the contents.

no characters from the clipboard are visible whatsoever and yet,t the file is written 

### Outcome

the contents of the wl-clipboard should be visible at all times on the minibufffer

if pasting say `foo-'bar-baz'-foo` the contents should not only be visible on the minibuffer but also redirected if so desired with any `mv` operation right after 

### Expectations

copying  `foo-'bar-baz'-foo` should paste the contents on the minibuffer and not just writing for the sake of it without either a prompt or only until after `<ret>` follows 

anything less than that is just a crappy and sloppy implementation that was not there before

**Comment by Screwtapello** [comment]
I don't understand the reproduction steps you've provided.

When you say "copy", do you mean `/bin/cp`, or yanking text from within Kakoune, or copying text to the system clipboard?

When you say "on the scratch buffer paste the contents", do you mean Kakoune's `p` command, or your terminal's "paste from system clipboard" command (often bound to Shift-Insert, or Ctrl-Shift-V)?

When you say "yet the file is written", what file are we talking about? The scratch buffer doesn't represent a file and can't be written.

Are you using some system clipboard integration plugin?

**Comment by nonumeros** [comment]
see the correction above 

`:w ` paste the wl-paste contents of `foo-'bar-baz'-foo` there

**Comment by nonumeros** [comment]
Have to close it. Sorry for the noise. 

I have a few versions installed and cannot reproduce with the latest head. `wl-paste` pastes the contents successfully. I would have to compile again each and every kak version I have to find out the culprit of pasting right up into the minibuffer the contents even before a `<ret>` operation is registered .

sorry for the noise again. The issue is there but it's time consuming for me to compile each kak version I have.

**Comment by nonumeros** [comment]
@Screwtapello  I felt obligated to find out the version/revision number where the above issue with the `wl-paste` happened. 

 I had compiled kak (latest head) on a different system, and I simply forgot (or mixed up) the two versions when I reported this issue. I could have sworn that it was the latest dated as stated on 4-25-2024. But I was wrong. 

It 's actually https://github.com/mawww/kakoune/commit/98b84f2b05e8532703898a85abd06fc28c4e40ac 

my apologies
